### ISSUE TITLE
Possible Enhancements For Reroll (1-6)

### ISSUE THREAD

**Issue by AbsorbingSafe11** [SmashingMods/Reroll#13]
1) Can the Reroll be activated on death?
1a) If this is implemented can the KeepInventory game rule be used, if it is on?

2) Can the inventory be put into a chest (possibly double chest) instead of being destroyed? (to possibly be found much later, if KeepInventory was off and no Grave mod was detected)

3) Can integration with with map mods to erase map data and waypoints? (like FTB Chunks, Xeno's maps, and Journeymap)
3a)  Can Chunk claims possibly  be erased as well? (not high priority, since they are team based frequently)

4) Can Ender chests be effected?

5) Can Soulbound items be effected? (including the various sorts, since there are like 20 different types)

6) Can Config options be added for the above? (as modpacks will want different things and most players will like more options anyways)

7) Can a 1.18.1 version be made? (no need to rush as 1.18.1 and 1.18.2 are still a bit unstable at time of writing)

**Comment by AbsorbingSafe11** [comment]
Couldn't figure out how to add any labels, so anyone who knows how, please add the Enhancement label.
Also, if any of these get added I will try to update the issue and mention which ones got added.

**Comment by Dark-Arcana** [comment]
You did everything just fine. Thanks for adding it here. I will need to make an issue template that makes it easier for people to know what to post.

**Comment by Dark-Arcana** [comment]
Original reply to your comment on CurseForge for context:

---

1) This is a great idea and I will probably add it.
1a) Yes.

2) This is possible too, maybe a little difficult.

3) That's mod integration. I can probably do that. I'm planning to add integration as I go.

3a) Depends on the mod and how it can be interacted with, and the version of the game.

4) Ender chests are wiped currently. There is no config, but it's not a bad option. Might add it!

5) Not yes, but let's add it.

6) I originally made this for an Akliz partner for the modpack Trepidation specifically to solve a problem. But this mod is actually a unique and interesting idea, so I plan on porting it to 1.16.5 and then 1.18.1. First I want to make sure the functionality in 1.12 is pretty good before moving on.

7) Yeah! The world is literally infinite. We shouldn't have to make an entirely new world when we want to start over.

**Comment by Dark-Arcana** [comment]
Alright I've decided to only add a few things in your current list for 1.12. I'm not going to support any map or gravestone mods.

I've added an option to put your inventory into a double chest where you died. It's kinda funky, but I'm going to improve this in 1.16.5 or newer, not 1.12.2. Ender chest has a config option now too.

Future versions will support all of the things on your list so keep an eye out.
